app/(auth)/change-password/page.tsx

```diff
@@ -0,0 +1,100 @@
+'use client'
+import { useState } from 'react'
+import { useForm } from 'react-hook-form'
+import {
+  Box,
+  Flex,
+  FormLabel,
+  FormErrorMessage,
+  FormControl,
+  Input,
+  Heading,
+  useToast,
+  useBoolean
+} from '@chakra-ui/react'
+import { useRouter } from 'next/navigation'
+import {
+  changePasswordResolver,
+  ChangePasswordSchema
+} from '@/(auth)/change-password/schema'
+import { PrimaryButton } from '@/components/button'
+
+export default function ChangePassword() {
+  const toast = useToast()
+  const [isLoading, setIsLoading] = useBoolean()
+  const [toastId, setToastId] = useState<ReturnType<typeof toast> | undefined>(
+    undefined
+  )
+  const router = useRouter()
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors }
+  } = useForm<ChangePasswordSchema>({
+    resolver: changePasswordResolver
+  })
+
+  const changePasswordHandler = handleSubmit(
+    async (data: ChangePasswordSchema) => {
+      setIsLoading.on()
+      const response = await fetch('/change-password/action', {
+        method: 'POST',
+        body: JSON.stringify(data)
+      })
+      setIsLoading.off()
+      if (response.ok) {
+        router.push('/')
```

Comment:
Replaced server action with api router, cuz I wanted to use response data. But apparently, `server action` can't respond data to `client component` written in another file.

Reply:
Hmm I think you can! With action method not onSubmit and wrapping with another function.
```
 <form action={onAction} >

  const onAction: () => void = handleSubmit(
    onSubmitFunc
  );
```

If you use POST method, maybe the file name is better like `api/route.ts`

Reply:
> Kana question:
> If you use POST method, maybe the file name is better like api/route.ts
> Yoshi answer:
> So how about /account/password/page.tsx for UI and /account/password/update/route.ts ?

I prefer to Yoshi's approach! The approach is close feature base rather than current change-password path.

Reply:
> But I didn't get a response and it was hard to handle error with following reasons.

@Yo-mah-Ya Did you see any of the following errors?

![Screenshot 2024-01-13 at 23 51 35](https://github.com/tabi-memo/frontend/assets/7115171/f0c32b60-630c-4fa0-9e41-33a2a3ac3e48)

Reply:
>If you use POST method, maybe the file name is better like api/route.ts
api/route.ts seems that every api calls goes to this file and app router doesn't have to do with api router.
So how about /account/password/page.tsx for UI and /account/password/update/route.ts ? The reason why I suggest this is that this is part of the features in account page, so that page should be the child of it, and page.tsx and route.tsx are can't be put in a same directory.

OK!